consensus/enclave/impl/src/lib.rs

```diff
@@ -322,6 +322,21 @@ impl SgxConsensusEnclave {
         Ok(transactions)
     }
 
+    /// Validate the provided timestamp is valid for use in building a block.
+    ///
+    /// # Arguments
+    /// * `timestamp` - The timestamp to validate
+    /// * `parent_block` - The parent block of the block being built.
+    fn validate_timestamp(timestamp: u64, parent_block: &Block) -> Result<u64> {
```

Comment:
bringing up the comment from, https://github.com/mobilecoinfoundation/mobilecoin/pull/3711#discussion_r1393478335
> might want to require timestamp=0 if the block version does not support timestamps, and only if it does ensure that the timestamp is greater than (or greater-than or equal) to the parent.

Do we want to keep supporting older block versions in consenus? Should we have some kind of logic that says consensus will only consense on a block version 4 or higher?
We have logic to ensure the block version doesn't go down.

Reply:
Older block versions are supported in https://github.com/mobilecoinfoundation/mobilecoin/pull/3733/commits/5a107ab9d09b3a93e4354940ea8e61b998adbd29